### PR TITLE
fix: overflow in segmented images

### PR DESCRIPTION
**Pull request by hollandjg**
the TIFF versions of images with more than 255 floes overflow. This stops them being coerced to 8 bits, and instead allows these images to have higher bit depth and be able to represent more floes.

### CODE REVIEW

File: src/ebfloeseg/preprocess.py [hunk 1]
```diff
@@ -243,18 +244,20 @@ def _preprocess(
 
     # saving the label floes tif
     fname = "final.tif"
+    assert output.min() >= 0
```
Comment (cpaniaguam) [inline src/ebfloeseg/preprocess.py:247]: What is this assert statement for? Did you forget to delete it or did you intend to throw an exception?

File: src/ebfloeseg/savefigs.py [hunk 2]
```diff
@@ -31,15 +35,32 @@ def imsave(
     else:
         fname = save_direc / fname
 
+    if rollaxis:
+        img = np.rollaxis(img, axis=2)
+        axis = None
+    else:
+        axis = 1
+
+    if img.dtype == np.dtype("bool") or dtype == np.dtype("bool"):
```
Comment (cpaniaguam) [inline src/ebfloeseg/savefigs.py:44]:
```suggestion
    if np.issubdtype(img.dtype, np.bool_) or np.issubdtype(dtype, np.bool_):
```

File: src/ebfloeseg/utils.py [hunk 3]
```diff
@@ -191,3 +191,68 @@ def get_wcuts(red_masked):
...
+        raise NotImplementedError(
+            "%s not supported, dtype must be integer" % str(arr.dtype)
+        )
+
+    arr_min = arr.min()
+    arr_max = arr.max()
+
+    for dtype in list(map(np.dtype, ["u1", "i1", "u2", "i2", "u4", "i4", "u8", "i8"])):
+        if (arr_min >= np.iinfo(dtype).min) and (arr_max <= np.iinfo(dtype).max):
+            return dtype
```
Comment (cpaniaguam) [inline src/ebfloeseg/utils.py:254]: I think map objects are iterable.

```suggestion
    int_types = ["u1", "i1", "u2", "i2", "u4", "i4", "u8", "i8"]
    for dtype in map(np.dtype, int_types):
        dtinfo = np.iinfo(dtype)
        if (arr_min >= dtinfo.min) and (arr_max <= dtinfo.max):
            return dtype
```

File: src/ebfloeseg/preprocess.py [hunk 4]
```diff
@@ -140,7 +141,7 @@ def _preprocess(
             fname=fname,
             count=1,
             rollaxis=False,
-            as_uint8=True,
+            dtype=np.dtype("bool"),
```
Comment (cpaniaguam) [inline src/ebfloeseg/preprocess.py:144]:
```suggestion
            dtype=np.bool_,
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Thanks for this. A couple little things to consider.

File: src/ebfloeseg/preprocess.py (hunk 1 above)
Comment (hollandjg) [reply src/ebfloeseg/preprocess.py:247]: I intend to throw an exception if this isn't fulfilled – it should never occur, but it'll result in errors if it does.

**Review by cpaniaguam — APPROVED** [review]
lgtm

**Pull request merged**